pool/hub.go

```diff
@@ -562,72 +562,27 @@ func (h *Hub) Run(ctx context.Context) {
 	h.shutdown()
 }
 
-// ClientInfo represents client miner information.
-type ClientInfo struct {
-	Miner    string
-	IP       string
-	HashRate *big.Rat
-}
-
 // FetchClientInfo returns connection details about all pool clients.
-func (h *Hub) FetchClientInfo() map[string][]*ClientInfo {
-	clientInfo := make(map[string][]*ClientInfo)
+func (h *Hub) FetchClientInfo() []*Client {
```

Comment:
If you're going to maintain the modifications to this function then the function signature will have to change some more. Since it's returning a slice of clients now I don't think it should be called `FetchClientInfo` anymore, perhaps `FetchHubClients` or `FetchClients` would do. Also the name `allClients` seems unnecessary here, you could call the client slice `clients` and enumerate the endpoint with `for _, c := range endpoint.clients`, this way you can append to the slice like this: `clients = append(clients, c)`. I think this is readable without being verbose.